lib/logstash/plugin_mixins/http_client_ssl_configs.rb

```diff
@@ -0,0 +1,212 @@
+      # If you need to use a custom keystore (`.jks`) specify that here. This does not work with .pem keys!
+      base.config :keystore, :validate => :path, :deprecated => 'Use `ssl_keystore_path` instead'
+      # Specify the keystore password here.
+      # Note, most .jks files created with keytool require a password!
+      base.config :keystore_password, :validate => :password, :deprecated => 'Use `ssl_keystore_password` instead'
+      # Specify the keystore type here. One of `JKS` or `PKCS12`. Default is `JKS`
+      base.config :keystore_type, :validate => :string, :default => 'JKS', :deprecated => 'Use `ssl_keystore_type` instead'
+      # If you need to use a custom truststore (`.jks`) specify that here. This does not work with .pem certs!
+      base.config :truststore, :validate => :path, :deprecated => 'Use `ssl_truststore_path` instead'
+      # Specify the truststore password here.
+      # Note, most .jks files created with keytool require a password!
+      base.config :truststore_password, :validate => :password, :deprecated => 'Use `ssl_truststore_password` instead'
+      # Specify the truststore type here. One of `JKS` or `PKCS12`. Default is `JKS`
+      base.config :truststore_type, :validate => :string, :default => 'JKS', :deprecated => 'Use `ssl_truststore_type` instead'
+      # NOTE: the default setting [] uses Java SSL engine defaults.
+      base.config :ssl_supported_protocols, :validate => %w[TLSv1.1 TLSv1.2 TLSv1.3], :default => [], :list => true
+      # Naming aligned with the Elastic stack.
+      #   full: verifies that the provided certificate is signed by a trusted authority (CA) and also verifies that the
+      #         server’s hostname (or IP address) matches the names identified within the certificate
+      #   none: no verification of the server’s certificate
+      base.config :ssl_verification_mode, :validate => %w[full none], :default => 'full'
+      # The list of cipher suites to use, listed by priorities.
+      # Supported cipher suites vary depending on which version of Java is used.
+      base.config :ssl_cipher_suites, :validate => :string, :list => true
+    end
+
+    private_class_method :setup_http_client_deprecated_ssl_configs
+
+    public
+
+    # Normalizes the deprecated SSL options, linking the old settings to the standardized ones.
+    # To use this method, all standardized options should be already registered.
+    # @raise [ArgumentError] if missing or invalid configurations are supplied
+    # @raise `LogStash::ConfigurationError` if conflicting params are supplied
+    # @return [void]
+    def normalize_http_client_deprecated_configs!
+      @ssl_certificate_authorities = normalize_config(:ssl_certificate_authorities) do |normalize|
+        normalize.with_deprecated_mapping(:cacert) do |cacert|
+          [cacert]
+        end
+      end
+
+      params['ssl_certificate_authorities'] = @ssl_certificate_authorities unless @ssl_certificate_authorities.nil?
```

Comment:
It's a safeguard, I've seen a few plugins strongly relying on the `param['..']` values instead of vars, In that case, when the plugin is being normalized, it doesn't need to check both (deprecated and new) parameter to get the value.